### ISSUE TITLE
Bug: 0.8.3alpha1 crashes when hotkey is pressed, but only on some laptops

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#251]
Whenever a hotkey is pressed, it dies with a weird stack trace that I don't understand (see below). 
Never crashes on my laptop, running same OS X (10.3), only on Regina's laptop. 
Normal menu keys will not crash, it's only hotkeys.  I was just in there adding a preference, so I'll try backing that out.

```
Process:               SquareDeskPlayer [15430]
Path:                  /Applications/SquareDeskPlayerPrevious.app/Contents/MacOS/SquareDeskPlayer
Identifier:            Zenstar.SquareDeskPlayer
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           SquareDeskPlayer [15430]
User ID:               501

Date/Time:             2017-12-29 15:26:33.798 -0800
OS Version:            Mac OS X 10.13.2 (17C88)
Report Version:        12
Anonymous UUID:        C882D40A-CEDE-D275-B38C-80D69CFA7B9C

Sleep/Wake UUID:       20C78612-C4B9-48C4-A54E-7C14DBD9A197

Time Awake Since Boot: 250000 seconds
Time Since Wake:       910 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

VM Regions Near 0:
--> 
    __TEXT                 0000000105658000-00000001058d5000 [ 2548K] r-x/rwx SM=COW  /Applications/SquareDeskPlayerPrevious.app/Contents/MacOS/SquareDeskPlayer

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   Zenstar.SquareDeskPlayer      	0x00000001056ffaa5 0x105658000 + 686757
1   Zenstar.SquareDeskPlayer      	0x00000001056a3405 0x105658000 + 308229
2   Zenstar.SquareDeskPlayer      	0x0000000105727e1c 0x105658000 + 851484
3   Zenstar.SquareDeskPlayer      	0x0000000105728c68 0x105658000 + 855144
4   org.qt-project.QtCore         	0x0000000106936d9b QMetaObject::activate(QObject*, int, int, void**) + 2027
5   org.qt-project.QtWidgets      	0x0000000105b83e4f QAction::activate(QAction::ActionEvent) + 303
6   org.qt-project.QtCore         	0x0000000106936923 QMetaObject::activate(QObject*, int, int, void**) + 883
7   org.qt-project.QtCore         	0x000000010692f9ec QObject::event(QEvent*) + 156
8   org.qt-project.QtWidgets      	0x0000000105b8cedd QApplicationPrivate::notify_helper(QObject*, QEvent*) + 269
9   org.qt-project.QtWidgets      	0x0000000105b8f822 QApplication::notify(QObject*, QEvent*) + 5906
10  org.qt-project.QtCore         	0x0000000106905aa4 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 164
11  org.qt-project.QtCore         	0x00000001069066d8 QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) + 600
12  libqcocoa.dylib               	0x0000000108b2ed6e 0x108b08000 + 159086
13  libqcocoa.dylib               	0x0000000108b2f631 0x108b08000 + 161329
14  com.apple.CoreFoundation      	0x00007fff3efd25a1 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
15  com.apple.CoreFoundation      	0x00007fff3f08c35c __CFRunLoopDoSource0 + 108
16  com.apple.CoreFoundation      	0x00007fff3efb5040 __CFRunLoopDoSources0 + 208
17  com.apple.CoreFoundation      	0x00007fff3efb44bd __CFRunLoopRun + 1293
18  com.apple.CoreFoundation      	0x00007fff3efb3d23 CFRunLoopRunSpecific + 483
19  com.apple.HIToolbox           	0x00007fff3e2cbe26 RunCurrentEventLoopInMode + 286
20  com.apple.HIToolbox           	0x00007fff3e2cba9f ReceiveNextEventCommon + 366
21  com.apple.HIToolbox           	0x00007fff3e2cb914 _BlockUntilNextEventMatchingListInModeWithFilter + 64
22  com.apple.AppKit              	0x00007fff3c596f5f _DPSNextEvent + 2085
23  com.apple.AppKit              	0x00007fff3cd2cb4c -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 3044
24  com.apple.AppKit              	0x00007fff3c58bd6d -[NSApplication run] + 764
25  libqcocoa.dylib               	0x0000000108b2e53f 0x108b08000 + 156991
26  org.qt-project.QtCore         	0x0000000106901eb1 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 401
27  org.qt-project.QtCore         	0x0000000106906115 QCoreApplication::exec() + 341
28  Zenstar.SquareDeskPlayer      	0x000000010565f4f8 0x105658000 + 29944
29  Zenstar.SquareDeskPlayer      	0x000000010565f184 0x105658000 + 29060

```

**Comment by mpogue2** [comment]
I got it to crash once on a fresh laptop (same OS X = 10.3), but now I can't get it to crash at all.
This is pretty weird.  Two laptops, same OS, no crashes, and third laptop crashes easily.

**Comment by mpogue2** [comment]
OK, I think I know what was happening.  This comment in the key handling code was a big clue:
"// EXPERIMENTAL FIX FIX FIX, will crash if webview[n] not exist yet".

On Regina's laptop, it was the first time she was running the 0.8.3alpha1 version of SquareDesk, so she did not have any Reference tab items.  Therefore, webview[0] was NULL, and the first attempt to access it (whenever a hotkey is pressed), resulted in a crash.  On my dev laptop, I have several Reference tab items, so no crash.  On my test laptop, it crashed only until I did the Dropbox update, which brought in some Reference materials, and then it no longer crashed.  Thus the weird inconsistent behavior.

I added a new function "someWebViewHasFocus()", which checks only the webview[i] that are valid, based on the numWebviews counter.  I'll ask Regina to try it out again!

Commit: 16d7f607e02c20cb4a789d4218c41dd29e276bb3 .
